Content.Server/Salvage/SalvageSystem.cs

```diff
@@ -0,0 +1,310 @@
+
+        [ViewVariables]
+        public float StateTimer = 0.0f;
+
+        public const float PullInTimer = 2.0f;
+        public const float HoldTimer = 240.0f;
+        public const float LeaveTimer = 60.0f;
+        public const float AngularVelocityRangeRadians = 0.25f;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<RoundRestartCleanupEvent>(Reset);
+            SubscribeLocalEvent<SalvageMagnetComponent, InteractHandEvent>(OnInteractHand);
+            SubscribeLocalEvent<SalvageMagnetComponent, ExaminedEvent>(OnExamined);
+        }
+
+        private void OnExamined(EntityUid uid, SalvageMagnetComponent comp, ExaminedEvent args)
+        {
+            if (!args.IsInDetailsRange)
+                return;
+
+            switch (State)
+            {
+                case SalvageSystemState.Inactive:
+                    args.PushMarkup(Loc.GetString("salvage-system-magnet-examined-inactive"));
+                    break;
+                case SalvageSystemState.PullingIn:
+                    args.PushMarkup(Loc.GetString("salvage-system-magnet-examined-pulling-in"));
+                    break;
+                case SalvageSystemState.Active:
+                    args.PushMarkup(Loc.GetString("salvage-system-magnet-examined-active", ("timeLeft", Math.Floor(StateTimer))));
+                    break;
+                case SalvageSystemState.LettingGo:
+                    args.PushMarkup(Loc.GetString("salvage-system-magnet-examined-letting-go"));
+                    break;
+            }
+        }
+
+        private void Reset(RoundRestartCleanupEvent ev)
+        {
+            PulledObject = EntityUid.Invalid;
+            State = SalvageSystemState.Inactive;
+            StateTimer = 0.0f;
+        }
+
+        private void OnInteractHand(EntityUid uid, SalvageMagnetComponent sm, InteractHandEvent args)
+        {
+            if (args.Handled)
+                return;
+            args.Handled = true;
+            _popupSystem.PopupEntity(CallSalvage(), uid, Filter.Entities(args.User.Uid));
+        }
+
+        private bool TryGetSalvagePlacementLocation(out MapCoordinates coords, out Angle angle)
+        {
+            coords = MapCoordinates.Nullspace;
+            angle = Angle.Zero;
+            foreach (var (smc, tsc) in EntityManager.EntityQuery<SalvageMagnetComponent, TransformComponent>(true))
+            {
+                coords = new EntityCoordinates(smc.OwnerUid, smc.Offset).ToMap(EntityManager);
+                var grid = tsc.GridID;
+                if (grid != GridId.Invalid)
+                {
+                    // Has a valid grid - synchronize angle so that salvage doesn't have to deal with cross-grid manipulation issues
+                    angle = _mapManager.GetGrid(grid).WorldRotation;
+                }
+                return true;
+            }
+            return false;
+        }
```

Comment:
This should really be finding somewhere not near the station instead of using a hardcoded offset.

Reply:
It lets the map designer pick where the magnet machine on the station spawns the salvage. It's currently set to somewhere above where the magnet is. It's in a good spot, because you can get to it from the EVA exit.